internal/controllers/operator_controller_test.go

```diff
@@ -954,75 +903,64 @@ var _ = Describe("Operator Controller Test", func() {
 				Expect(cond.Message).To(Equal("installation has not been attempted as resolution failed"))
 			})
 		})
-		AfterEach(func() {
-			verifyInvariants(ctx, operator)
+		When("an invalid semver is provided that bypasses the regex validation", func() {
+			var (
+				pkgName    string
+				fakeClient client.Client
+			)
+			BeforeEach(func() {
+				opKey = types.NamespacedName{Name: fmt.Sprintf("operator-validation-test-%s", rand.String(8))}
 
-			err := cl.Delete(ctx, operator)
-			Expect(err).To(Not(HaveOccurred()))
-		})
-	})
-	When("an invalid semver is provided that bypasses the regex validation", func() {
```

Comment:
Moved this block into the `When("the operator exists")` so that it also passes through the `verifyInvariants` check that we apply to operators after calling `Reconcile`.